pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -181,3 +207,37 @@ func (c *Cluster) enablePrometheusModule(clusterName string) error {
 	}
 	return nil
 }
+
+func (c *Cluster) enableServiceMonitor(service *v1.Service) error {
+	name := service.GetName()
+	namespace := service.GetNamespace()
+	serviceMonitor, err := k8sutil.GetServiceMonitor(serviceMonitorFilePath)
+	if err != nil {
+		return err
+	}
+	serviceMonitor.SetName(name)
+	serviceMonitor.SetNamespace(namespace)
+	owner := []metav1.OwnerReference{c.ownerRef}
+	serviceMonitor.SetOwnerReferences(owner)
+	if _, err := k8sutil.CreateOrUpdateServiceMonitor(serviceMonitor); err != nil {
+		return fmt.Errorf("failed to enable servicemonitor. %+v", err)
+	}
+	return nil
+}
+
+func (c *Cluster) deployPrometheusRule(name, namespace string) error {
+	version := "v" + strconv.Itoa(c.clusterInfo.CephVersion.Major)
+	prometheusRuleFilePath = strings.Replace(prometheusRuleFilePath, "VERSION", version, 1)
+	prometheusRule, err := k8sutil.GetPrometheusRule(prometheusRuleFilePath)
+	if err != nil {
+		return err
+	}
+	prometheusRule.SetName(name)
+	prometheusRule.SetNamespace(namespace)
+	owner := []metav1.OwnerReference{c.ownerRef}
+	prometheusRule.SetOwnerReferences(owner)
```

Comment:
Note that we need #3350 to merge so we can set the ownerreferences on OpenShift

Reply:
Should we add `rook` as the owner or `prometheus operator` ?

Reply:
It should be the cephcluster that is the owner. Then if the cephcluster CR is deleted, this resource will automatically be deleted. However, then we have the question about when to clean up if the rule is written to another namespace. I believe the answer lies in setting multiple owner references in that case. If the rules already exist in the other namespace, this cephcluster should also be added to its owner references. Then the rules should automatically be deleted after the last cephcluster is deleted.

Reply:
Having multiple owners would work when we have a single `prometheusRule` object (say in `openshift-monitoring`) for all the clusters in different namespaces.
But, if we have `prometheusRule` object in each namespace, we could have each cephcluster as owner. Then,
- multiple objects may bring same set of rules
- if one cluster is dependent on rules belonging to different cluster (without adding ownership), cleanup could be messier.

Reply:
A couple questions:
- Prometheus would allow the same rules in different namespaces, right? 
- If the rules are shared for openshift-monitoring namespace by multiple clusters, the rook operator should add each cephcluster to the ownerreferences, right? All cleanup would be based on the ownerreferences. What is the messy part?

Reply:
> A couple questions:
> 
>     * Prometheus would allow the same rules in different namespaces, right?
Yes. But, there is no deduplication.
Meaning, UI will list a rule (say `cephMgrAbsent`) `n` times if it is present in `n` namespaces. This also might lead to `n` alerts for same issue.
> 
>     * If the rules are shared for openshift-monitoring namespace by multiple clusters, the rook operator should add each cephcluster to the ownerreferences, right? All cleanup would be based on the ownerreferences. What is the messy part?
If I have a set of `ceph` rules : `rule-1` deployed by `cluster-1`(owner for rule-1) and `rule-2` deployed by `cluster-2`(owner for rule-2 only). For avoiding duplicates `rule-2` doesn't include the set of rules in `rule-1`.
Now, if `cluster-1` is cleaned, `rule-1` will be lost (cluster-2 in not in its list of owners). `cluster-2` will have to deploy its own set of `rule-1`.
This cluster and rules relationship is what I see complex.

Reply:
My thinking is that we should restrict it to a single namespace (probably the namespace where prometheus is deployed). And, we will have a single set of rules for all v14 cluster. Another set of rules for v15 clusters and so on. 
Each cluster could add itself as owner to these same rules based on its own version.

Reply:
Supporting different set of rules for same versions of ceph cluster in different namespaces looks anti-pattern.

Reply:
Understood that we will need to use a single namespace for the rules, then add the ownerreferences for each cluster. For upstream, we will just have the default be the same namespace, and we would need to document that if they are configuring multiple clusters they should use a single namespace.

Reply:
Ack. Added a Comment in `cluster.yaml` as `Recommended:`. Will that do or do we need an explicit documentation?